### PR TITLE
Test all routes and core functionality of the application

### PR DESCRIPTION
**Pull request by JayKay24**
#### What does this PR do?

This PR incorporates tests for the core functionality of the application.

#### What are the relevant pivotal tracker stories?
[#159071954](https://www.pivotaltracker.com/story/show/159071954)

### CODE REVIEW

File: app/controllers/category_controller.rb [hunk 1]
```diff
@@ -3,22 +3,22 @@
...
+  get '/category/new' do
+    check_if_logged_in
+
     haml :'category/create_category'
   end
 
   get '/categories' do
-    redirect '/login' unless current_user
-    @current_user = current_user
+    check_if_logged_in
```
Comment (morriskimani) [inline app/controllers/category_controller.rb:13]: I like that you've made this logic into a function. However, this is still a bit repetitive. It appears at the beginning of each route handler. Why not leverage **before filters**

File: app/controllers/user_controller.rb [hunk 2]
```diff
@@ -3,6 +3,19 @@
...
 
 class UserController < ApplicationController
+  def get_user_params
+    user_params = {
+      first_name: params[:first_name],
+      last_name: params[:last_name],
+      email: params[:email]
+    }
+    if params[:password]
+      password_match = /^(?=.*[a-zA-Z])(?=.*[0-9])(?=.*[\W]).{8,}$/.match?(params[:password])
```
Comment (morriskimani) [inline app/controllers/user_controller.rb]: I thought we have this validation in the user model. Why do we need it here?

File: services/bing_service.rb [hunk 3]
```diff
@@ -5,7 +5,7 @@
 module BingService
   class BingWebSearchApi
     def links(search_term)
-      access_key = ENV['BING_API_ACCESS_KEY']
+      access_key = 'ff874699981347b49bb78f71f55b63d5'
```
Comment (morriskimani) [inline services/bing_service.rb:8]: Why are we exposing app secrets like this?

File: spec/helpers/controller_helpers.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+module Helpers
+  module Controllers
+    def app
+      @app ||= described_class
+    end
+
+    def expect_redirection_to(route)
```
Comment (morriskimani) [inline spec/helpers/controller_helpers.rb:7]: 👍 
Good use of helpers

File: spec/controllers/category_controller_spec.rb [hunk 5]
```diff
@@ -1,19 +1,41 @@
...
 
 RSpec.describe CategoryController do
-  def app
-    @app ||= CategoryController
+  include Helpers::Controllers
+
+  DatabaseCleaner.strategy = :truncation
+
+  before :each do
+    @user = create(:user)
```
Comment (morriskimani) [inline spec/controllers/category_controller_spec.rb:13]: Prefer using `let` vs instance variables inside your tests. [See some reasons why](https://www.ombulabs.com/blog/rails/rspec/ruby/let-vs-instance.html)

**Review by morriskimani — COMMENTED** [review]
Good stuff!
Just some minor comments

File: app/controllers/user_controller.rb (hunk 2 above)
Comment (JayKay24) [reply app/controllers/user_controller.rb]: I removed it. Was not supposed to be there.

**Pull request merged**